modules/tinymce/src/core/test/ts/browser/api/dom/RangeExpandTest.ts

```diff
@@ -0,0 +1,91 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Unicode } from '@ephox/katamari';
+import { TinyHooks, TinySelections } from '@ephox/mcagar';
+import { assert } from 'chai';
+
+import RangeUtils from 'tinymce/core/api/dom/RangeUtils';
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.core.api.dom.RangeExpandTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    indent: false,
+    base_url: '/project/tinymce/js/tinymce'
+  }, []);
+
+  const textEntry = 'A B ' + Unicode.nbsp + 'C D';
+
+  const makeBaseSelection = (editor: Editor, startOffset: number, endOffset: number, nodeOffset: number = 0) => {
+    const body = editor.getBody();
+    const range = editor.dom.createRng();
+    const node = body.childNodes[0].childNodes[nodeOffset];
+    range.setStart(node, startOffset);
+    range.setEnd(node, endOffset);
+    return range;
+  };
+
+  const compareRanges = (range1: Range, range2: Range) => {
+    assert.equal(range1.startContainer, range2.startContainer, 'Mismatching start container');
+    assert.equal(range1.startOffset, range2.startOffset, 'Mismatching start offset');
+    assert.equal(range1.endContainer, range2.endContainer, 'Mismatching end container');
+    assert.equal(range1.endOffset, range2.endOffset, 'Mismatching end offset');
+  };
+
+  it('TINY-9001: The content is empty', () => {
+    const editor = hook.editor();
+    editor.setContent('');
+    TinySelections.setSelection(editor, [ 0 ], 0, [ 0 ], 0);
+    const startRange = editor.selection.getRng();
+    RangeUtils(editor).expand(startRange);
```

Comment:
Shouldn't this compare the return value. Since it's not mutating the input range but returning a new one. You could add a separate test that checks if it mutates I guess.
```suggestion
    const newRange = RangeUtils(editor).expand(startRange);
```